packages/blade/src/components/Input/BaseInput/useTaggedInput.ts

```diff
@@ -0,0 +1,167 @@
+} => {
+  const [activeTagIndex, setActiveTagIndex] = React.useState(-1);
+  const [inputValueUncontrolled, setInputValueUncontrolled] = React.useState(defaultValue ?? '');
+  const [tagsUncontrolled, setTagsUncontrolled] = React.useState<string[]>([]);
+  const isTagsControlled = Boolean(tags);
+
+  const getNewTagsArray = (indexToRemove: number): string[] => {
+    const currentTags = tags ?? tagsUncontrolled;
+
+    if (!currentTags) {
+      return [];
+    }
+
+    // Check if the index is valid
+    if (indexToRemove < 0 || indexToRemove >= currentTags.length) {
+      return currentTags; // Return the original array
+    }
+
+    // Create a new array without the element at the specified index
+    const newArray = currentTags
+      .slice(0, indexToRemove)
+      .concat(currentTags.slice(indexToRemove + 1));
+
+    return newArray;
+  };
+
+  const getTags = React.useMemo(
+    () => ({ size }: { size: NonNullable<BaseInputProps['size']> }): React.ReactElement[] => {
+      return getTagsGroup({
+        size,
+        tags: tags ?? tagsUncontrolled,
+        activeTagIndex,
+        isDisabled,
+        onDismiss: ({ tagIndex }) => {
+          console.log('dismiss', { tagIndex });
+          if (!isTagsControlled) {
+            setTagsUncontrolled(getNewTagsArray(tagIndex));
+          }
+          onTagChange?.({ tags: getNewTagsArray(tagIndex) });
+        },
+      });
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [activeTagIndex, tags, tagsUncontrolled, isDisabled],
+  );
+
+  const handleTaggedInputChange: FormInputOnEvent = ({ value }) => {
+    if (!isTaggedInput) {
+      return;
+    }
+    setInputValueUncontrolled(value ?? '');
+  };
+
+  const handleTagsClear = (): void => {
+    if (!isTaggedInput) {
+      return;
+    }
+
+    if (!isTagsControlled) {
+      setTagsUncontrolled([]);
+    }
+
+    onTagChange?.({ tags: [] });
+  };
+
+  const clearInput = (): void => {
+    const isControlledValue = value !== undefined;
+
+    if (isControlledValue) {
+      // In Controlled component, we don't clear input ourselves. We just call onChange with empty value
+      onChange?.({ name, value: '' });
+      return;
+    }
+
+    if (!inputRef.current) {
+      return;
+    }
+
+    setInputValueUncontrolled('');
+
+    if (isReactNative()) {
+      setTimeout(() => {
+        // eslint-disable-next-line @typescript-eslint/prefer-ts-expect-error
+        // @ts-ignore: clear does exist in react native
+        inputRef.current?.clear();
+      }, 10);
+      return;
+    }
+
+    inputRef.current.value = '';
+  };
+
+  const handleTaggedInputKeydown = (e: FormInputOnKeyDownEvent): void => {
+    if (!isTaggedInput) {
+      return;
+    }
+
+    const currentTags = tags ?? tagsUncontrolled;
+    const isControlledValue = value !== undefined;
+    const inputValue = isControlledValue ? value?.trim() : inputValueUncontrolled.trim();
+    if (e.key === 'Enter' || e.key === ',') {
```

Comment:
What about " " space? Do we not create tags on pressing SPACE?

Reply:
nope. the tag itself might have space at some places no 🤔